app/actions/client.py

```diff
@@ -11,6 +11,14 @@
 state_manager = IntegrationStateManager()
 
 
+class CollarData(pydantic.BaseModel):
+    collarID: str
+    collarType: str
+    comID: str
+    comType: str
+    key: str
+
```

Comment:
Tip: Using aliases would allow naming fields with snake_case to be `PEP8` compliant, and `.parse_obj()` would work with the camelCase field names. And `allow_population_by_field_name = True` will let `.parse_obj()` work with the snake_case field names too, just in case you need it.
```suggestion
class CollarData(BaseModel):
    collar_id: str = pydantic.Field(..., alias="collarID")
    collar_type: str = pydantic.Field(..., alias="collarType")
    com_id: str = pydantic.Field(..., alias="comID")
    com_type: str = pydantic.Field(..., alias="comType")
    key: str = pydantic.Field(..., alias="key")

    class Config:
        allow_population_by_field_name = True

```

Reply:
Also, since the `CollarData` model isn't used in `client.py`, and it seems to be more related to our own representation of collar data than to the vectronics API, we could move it out of client.py, into handlers.py.